docs/diagrams/commandhistory/CmdHistStorageSequenceDiagram.puml

```diff
@@ -0,0 +1,53 @@
+@startuml
+!include ../style.puml
+
+Participant ":Logic" as logic LOGIC_COLOR
+Participant ":Model" as model MODEL_COLOR
+Participant ":CommandHistoryStorage" as storage STORAGE_COLOR
+
+-[UI_COLOR]> logic : execute("help")
+activate logic LOGIC_COLOR
+
+logic -[LOGIC_COLOR]> logic : parseAndExecute("help")
+activate logic LOGIC_COLOR
+
+logic -[LOGIC_COLOR]-> logic
+deactivate logic
+
+opt command executes successfully
```

Comment:
Should this part be reflected via an optional frame, or should the fact that it was successfully executed be an underlying assumption? Maybe the branch here is better explained through an activity diagram, and the sequence diagram should show the "Happy path"? What do you think?

Reply:
Hmm good point. An optional frame might suggest that a command being unsuccessful is the "happy path". My original intention was to convey the fact that unsuccessful commands will not be recorded in the command history, but I guess an activity diagram or a note in the sequence diagram might be more appropriate.

Thank you, I will consider this suggestion seriously for a future update.